core/modules/loader/BufferUdp.h

```diff
@@ -0,0 +1,191 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ *
+ */
+#ifndef LSST_QSERV_LOADER_BUFFERUDP_H_
+#define LSST_QSERV_LOADER_BUFFERUDP_H_
+
+// system headers
+#include <arpa/inet.h>
+#include <cstring>
+#include <stdexcept>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include <boost/bind.hpp>
+#include <boost/asio.hpp>
+
+/// The absolute largest UDP message we would send.
+/// Usually, they should be much smaller.
+#define MAX_MSG_SIZE 6000
+
+
+namespace lsst {
+namespace qserv {
+namespace loader {
+
+class MsgElement;
+
+
+/// A buffer for reading and writing. Nothing can be read from the buffer until
+/// something has written to it.
+class BufferUdp {
```

Comment:
are there unit tests for this class? it seems like it would be valuable to have some.

Reply:
You mentioned in that file that it's an integration test. There is a lot of code here without testing at the 'unit' granularity, which I think is an issue.

Reply:
I made a separate ticket for creating unit tests DM-16649